docs/containerization-design.rst

```diff
@@ -0,0 +1,358 @@
+#. In order to satisfy `key Requirement #3 <#Key-Requirements-for-Containerization>`_, the execution environment
+   for flow pods will be constructed dynamically at run-time.
+
+   * Azkaban will provide a mechanism to dynamically select versions of components that constitute a functional
+     Azkaban Executor environment at dispatch time.
+   * Following this, a series of init containers will pull intended versions of various components to compose the
+     complete execution environment.
+   * The dynamic selection process will be utilized to provide canary capability for various Azkaban components.
+   * The design also introduces a few Admin APIs to make the task of image management easier.
+
+Design Details
+**************
+
+Image Management
+----------------
+* Azkaban will rely on docker images to create execution environment for flows. In order to satisfy
+  `key Requirement #3 <#Key-Requirements-for-Containerization>`_, the final container image will be constructed
+  dynamically using init-containers [1]_ when the flow pod is launched. The required layers will be
+  discovered as laid out in the `dispatch logic <#Dispatch-Logic>`_.
+* Azkaban execution environment is composed of the following types of dependencies:
+
++-----------------------+--------------------------------------------------------------------------+
+|    Dependency Type    |                          Description                                     |
++=======================+==========================================================================+
+| Platform Dependencies | Binaries/Configs/Secrets for Hadoop, Hive, Spark, Pig, Dali, Ksudo, etc. |
++-----------------------+--------------------------------------------------------------------------+
+| Azkaban Core          | Core Azkaban binaries/configs/secrets owned and managed by Azkaban       |
++-----------------------+--------------------------------------------------------------------------+
+| Azkaban JobTypes      | Binaries/Configs owned by JobType developers but are managed by Azkaban  |
+|                       | like KafkaPushJob, SparkJob, etc.                                        |
++-----------------------+--------------------------------------------------------------------------+
+
+* Azkaban Core forms the base docker image layered on top of a base image of choice: such as RHEL7.
+* Each of the above (Platform or jobtypes) will be packaged as a separate docker image layers on top of a simple
+  base image of choice. It is preferable to keep these images very small to optimize download times. Something
+  like a busybox/alpine image should suffice.
+* Individual images for job-types will allow independent development and release for the job-type developers without
+  any dependency on Azkaban. Here is an example image definition for KPJ (Kafka Push Job):
+
+.. code-block:: docker
+
+  FROM container-image-registry.corp.linkedin.com/rhel7-base-image/rhel7-base-image:0.16.9
```

Comment:
Perhaps remove any references to internal websites ?

Reply:
done.